src/applications/vaos/services/appointment/transformers.js

```diff
@@ -0,0 +1,273 @@
+    case APPOINTMENT_TYPES.ccAppointment:
+      return APPOINTMENT_STATUS.booked;
+    case APPOINTMENT_TYPES.ccRequest:
+    case APPOINTMENT_TYPES.request: {
+      return appointment.status === 'Cancelled'
+        ? APPOINTMENT_STATUS.cancelled
+        : APPOINTMENT_STATUS.pending;
+    }
+    case APPOINTMENT_TYPES.vaAppointment: {
+      const currentStatus = isVideoVisit(appointment)
+        ? appointment.vvsAppointments?.[0]?.status?.code
+        : appointment.vdsAppointments?.[0]?.currentStatus;
+
+      if (
+        (isPastAppointment &&
+          PAST_APPOINTMENTS_HIDE_STATUS_SET.has(currentStatus)) ||
+        (!isPastAppointment &&
+          FUTURE_APPOINTMENTS_HIDE_STATUS_SET.has(currentStatus))
+      ) {
+        return null;
+      }
+
+      const cancelled = CANCELLED_APPOINTMENT_SET.has(currentStatus);
+
+      return cancelled
+        ? APPOINTMENT_STATUS.cancelled
+        : APPOINTMENT_STATUS.booked;
+    }
+    default:
+      return APPOINTMENT_STATUS.booked;
+  }
+}
+/**
+ * Finds the datetime of the appointment depending on the appointment type
+ * and returns it as a moment object
+ *
+ * @param {Object} appt VAR appointment object
+ * @returns {Object} Returns appointment datetime as moment object
+ */
+function getMomentConfirmedDate(appt) {
+  if (isCommunityCare(appt)) {
+    const zoneSplit = appt.timeZone.split(' ');
+    const offset = zoneSplit.length > 1 ? zoneSplit[0] : '+0:00';
+    return moment
+      .utc(appt.appointmentTime, 'MM/DD/YYYY HH:mm:ss')
+      .utcOffset(offset);
+  }
+
+  const timezone = getTimezoneBySystemId(appt.facilityId)?.timezone;
+  const date = isVideoVisit(appt)
+    ? appt.vvsAppointments[0].dateTime
+    : appt.startDate;
+  return timezone ? moment(date).tz(timezone) : moment(date);
+}
+
+/**
+ * Returns url for user to join video conference
+ *
+ * @param {Object} appt VAR appointment object
+ * @returns {String} URL of video visit
+ */
+function getVideoVisitLink(appt) {
+  return appt.vvsAppointments?.[0]?.patients?.[0]?.virtualMeetingRoom?.url;
+}
+
+/**
+ * Returns appointment duration in minutes. The default is 60 minutes.
+ *
+ * @param {Object} appt VAR appointment object
+ * @returns
+ */
+function getAppointmentDuration(appt) {
+  const appointmentLength = parseInt(
+    appt.vdsAppointments?.[0]?.appointmentLength ||
+      appt.vvsAppointments?.[0]?.duration,
+    10,
+  );
+  return isNaN(appointmentLength) ? 60 : appointmentLength;
+}
+
+/**
+ * Builds participant and contained arrays for FHIR Appointment object which usually
+ * contain Location (Facility) and HealthcareService (Clinic) or video conference info
+ *
+ * @param {Object} appt  VAR appointment object
+ * @returns {Array} Array of participants of FHIR appointment
+ */
+function addParticipantAndContained(appt, transformed) {
+  const isVideo = isVideoVisit(appt);
+  let contained;
+
+  const participant = [
+    {
+      actor: {
+        reference: `Location/var${appt.facilityId}`,
```

Comment:
This facilityId is actually a site id and will map to an organization, rather than a location. But I'm not sure the right way to map this in the interim, because there won't be an Organization in the participant array. When we talk to EAS for real, we'll probably have to pull the full Location and HealthcareService data for each Appointment to get this id.

I think maybe for now put this id in a `legacyVAR` object on the Appointment, and we can make a ticket to figure out if we'll need both Location and HealthcareService resources and update the mapping accordingly.